doc/src/challenges/0007-atomic-types.md

```diff
@@ -0,0 +1,83 @@
+# Challenge 7: Safety of Methods for Atomic Types and `ReentrantLock`
+
+- **Status:** Open
+- **Tracking Issue:** [#83](https://github.com/model-checking/verify-rust-std/issues/83)
+- **Start date:** *2024-09-09*
+- **End date:** *2024-12-10*
+
+-------------------
+
+## Goal
+
+Verify the safety of:
+- The methods for atomic types in `core::sync::atomic` 
+- The implementation of `ReentrantLock` in `std::sync::reentrant_lock`
+
+### Success Criteria
+
+
+#### Part 1: Atomic Types
```

Comment:
Can you insert a sentence that explains why there are two parts here? At this point I was wondering whether this could equally be two separate challenges. Only later on you clarify that `ReentrantLock` relies on atomic types. I suggest that you make clear this dependency at this point.